packages/wallet-sdk/src/sign/scw/utils.ts

```diff
@@ -353,37 +360,39 @@ export async function waitForCallsTransactionHash({
 }
 
 export function createWalletSendCallsRequest({
-  to,
-  data,
-  value,
+  calls,
   from,
   chainId,
+  capabilities,
 }: {
-  to: Address;
-  data: Hex;
-  value: Hex;
+  calls: { to: Address; data: Hex; value: Hex }[];
   from: Address;
   chainId: Hex;
+  capabilities?: Record<string, unknown>;
 }) {
-  return {
+  const paymasterUrls = config.get().paymasterUrls;
+
+  let request: { method: 'wallet_sendCalls'; params: WalletSendCallsParameters } = {
     method: 'wallet_sendCalls',
     params: [
       {
         version: '1.0',
-        calls: [
-          {
-            to,
-            data,
-            value,
-          },
-        ],
+        calls,
         chainId,
         from,
         atomicRequired: true,
-        // TODO: Add paymaster capabilities from config
+        capabilities,
       },
     ],
   };
+
+  if (paymasterUrls?.[hexToNumber(chainId)]) {
```

Comment:
Should we validate that this is a hex before converting to a number?

Reply:
as a helper function i think this validation should happen somewhere else tbh

in all the use cases we're getting the chain id from a `PublicClient.chain.id` so we should not be seeing hex here.

seeing as it's a convenience method i've also updated the helper's type to be `number` instead and do the hex conversions inside where necessary